case-lib/hijack.sh

```diff
@@ -64,13 +64,10 @@ function func_exit_handler()
 
     # get ps command result as list
     local -a cmd_lst
-    # $$ as current script pid
-    # NOTICE: already test with $BASHPID:
-    # it can output the same result of $$
-    # but the result could not be stored in the array
-    readarray -t cmd_lst < <(pgrep -P $$ -a|grep -v "$SCRIPT_NAME")
+    # can't run pgrep in any subshell because the latter would pollute the list
+    if pgrep -P $$ -a > "$LOG_ROOT/children_left.txt"; then
```

Comment:
I do not see any need to use a tmp file here, this will bring more trouble with sub-test
why not use `readarray -t cmd_lst < <(pgrep -P $$ -a)`

And the whole logic is changed to me, the parent will still be killed by itself now.

How is `# can't run pgrep in any subshell because the latter would pollute the list` take effect here